UnicornRender/video/include/unicorn/video/Material.hpp

```diff
@@ -87,10 +87,38 @@ class Material
     /** @brief Returns color */
     UNICORN_EXPORT glm::vec3 GetColor() const;
 
+    /**
+     * @brief Sets texture cordinates
+     *
+     * @param x texture offset by x in pixels
+     * @param y texture offset by y in pixels
+     * @param width sprite width in pixels
+     * @param height sprite height in pixels
+     */
+    UNICORN_EXPORT void SetTextureCoordinates(int32_t x, int32_t y, int32_t width, int32_t height);
+
+    /**
+     * @brief Returns texture coordinates
+     *
+     * @return texture coordinates: {x, y, width, height}
+    */
+    UNICORN_EXPORT glm::vec4 GetTextureCoordinates() const;
```

Comment:
Semantically coordinates is the wrong word - you return x, y coordinates and width, height dimensions. For coordinates you would've used `{x0, y0, x1, y1}`
So this is more a `GetSpriteArea`